src/array_of_fixedsize.jl

```diff
@@ -12,11 +12,15 @@ call(::MaxFun, a, b) = max(a, b)
 call(::MinFun, a, b) = min(a, b)
 minimum{T <: FixedArray}(a::Vector{T}) = reduce(MinFun(), a)
 maximum{T <: FixedArray}(a::Vector{T}) = reduce(MaxFun(), a)
-function isapprox{FSA <: FixedArray, A <: Union{Array, FixedArray}}(a::FSA, b::A)
-    for i=1:length(a)
-        !isapprox(a[i], b[i]) && return false
-    end
-    true
+
+function isapprox{FSA <: FixedArray, A <: FixedArray}(x::FSA, y::A; rtol::Real=Base.rtoldefault(eltype(x),eltype(y)), atol::Real=0, norm::Function=vecnorm)
+    # Same behaviour as AbstractArray: julia/base/linalg/generic.jl
+	d = norm(x - y)
+    return isfinite(d) ? d <= atol + rtol*max(norm(x), norm(y)) : x == y
+end
+function isapprox{FSA <: FixedArray, A <: Array}(x::FSA, y::A; rtol::Real=Base.rtoldefault(eltype(x),eltype(y)), atol::Real=0, norm::Function=vecnorm)
+    d = norm(Array(x) - y) # operations between Arrays and FixedArrays not defined...
```

Comment:
+1

On 2 May 2016 at 14:42, Chris Foster <notifications@github.com> wrote:

> In src/array_of_fixedsize.jl
> <https://github.com/SimonDanisch/FixedSizeArrays.jl/pull/103#discussion_r61702009>
> :
>
> > @@ -12,11 +12,15 @@ call(::MaxFun, a, b) = max(a, b)
> >  call(::MinFun, a, b) = min(a, b)
> >  minimum{T <: FixedArray}(a::Vector{T}) = reduce(MinFun(), a)
> >  maximum{T <: FixedArray}(a::Vector{T}) = reduce(MaxFun(), a)
> > -function isapprox{FSA <: FixedArray, A <: Union{Array, FixedArray}}(a::FSA, b::A)
> > -    for i=1:length(a)
> > -        !isapprox(a[i], b[i]) && return false
> > -    end
> > -    true
> > +
> > +function isapprox{FSA <: FixedArray, A <: FixedArray}(x::FSA, y::A; rtol::Real=Base.rtoldefault(eltype(x),eltype(y)), atol::Real=0, norm::Function=vecnorm)
> > +    # Same behaviour as AbstractArray: julia/base/linalg/generic.jl
> > +	d = norm(x - y)
> > +    return isfinite(d) ? d <= atol + rtol*max(norm(x), norm(y)) : x == y
> > +end
> > +function isapprox{FSA <: FixedArray, A <: Array}(x::FSA, y::A; rtol::Real=Base.rtoldefault(eltype(x),eltype(y)), atol::Real=0, norm::Function=vecnorm)
> > +    d = norm(Array(x) - y) # operations between Arrays and FixedArrays not defined...
>
> Yeah, it seems safe if mixed operations would promote to the FSA involved.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/SimonDanisch/FixedSizeArrays.jl/pull/103/files/e3b4a84009e049a8e6bd677ffb1d14d3b04cc116#r61702009>
>